### PR TITLE
Spam Classification

### PR DESCRIPTION
**Pull request by ewdurbin**
Pursuant #2982

### CODE REVIEW

**Comment by ewdurbin** [comment]
@di @dstufft thoughts on the modeling thus far?

File: warehouse/spam/models.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            "reporter_user_id IS NOT NULL OR source = 'automation'",
+            name="valid_reporter_user_id",
+        ),
+    )
+
+    release_name = Column(Text, nullable=False)
+    release_version = Column(Text, nullable=False)
+    source = Column(report_source, nullable=False)
+    reporter_user_id = Column(UUID, nullable=True)
+    result = Column(Boolean, nullable=False)
```
Comment (di) [inline warehouse/spam/models.py]: Is this supposed to be some metric for "spaminess"?

Comment (ewdurbin) [reply warehouse/spam/models.py]: negative. in initial research of akismet and smythe the response to a classification request is a boolean. `True`/`False` or `verdict:BLOCK`/`verdict:ALLOW` (which can be trivially modeled as a Boolean)

**Comment by ewdurbin** [comment]
dope, I'll continue fleshing this out tonight.

**Pull request closed without merging**